packages/@aws-cdk/aws-lambda-python/lib/bundling.ts

```diff
@@ -16,137 +17,89 @@ export const BUNDLER_DEPENDENCIES_CACHE = '/var/dependencies';
-   * need to make sure it is updated every time the asset changes, or otherwise it is
-   * possible that some deployments will not be invalidated.
-   *
-   * @default - based on `assetHashType`
-   */
-  readonly assetHash?: string;
+  readonly architecture?: Architecture;
 }
 
 /**
  * Produce bundled Lambda asset code
  */
-export function bundle(options: BundlingOptions): lambda.Code {
-  const { entry, runtime, architecture, outputPathSuffix } = options;
-
-  const stagedir = cdk.FileSystem.mkdtemp('python-bundling-');
-  const hasDeps = stageDependencies(entry, stagedir);
-
-  const depsCommand = chain([
-    hasDeps ? `rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}` : '',
-    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}`,
-  ]);
-
-  // Determine which dockerfile to use. When dependencies are present, we use a
-  // Dockerfile that can create a cacheable layer. We can't use this Dockerfile
-  // if there aren't dependencies or the Dockerfile will complain about missing
-  // sources.
-  const dockerfile = hasDeps
-    ? 'Dockerfile.dependencies'
-    : 'Dockerfile';
-
-  // copy Dockerfile to workdir
-  fs.copyFileSync(path.join(__dirname, dockerfile), path.join(stagedir, dockerfile));
+export class Bundling implements CdkBundlingOptions {
+  public static bundle(options: BundlingProps): AssetCode {
+    return Code.fromAsset(options.entry, {
+      assetHash: options.assetHash,
+      assetHashType: options.assetHashType,
+      exclude: DEPENDENCY_EXCLUDES,
+      bundling: new Bundling(options),
+    });
+  }
 
-  const image = cdk.DockerImage.fromBuild(stagedir, {
-    buildArgs: {
-      IMAGE: runtime.bundlingImage.image,
-    },
-    platform: architecture.dockerPlatform,
-    file: dockerfile,
-  });
+  public readonly image: DockerImage;
+  public readonly command: string[];
 
-  return lambda.Code.fromAsset(entry, {
-    assetHashType: options.assetHashType,
-    assetHash: options.assetHash,
-    exclude: DEPENDENCY_EXCLUDES,
-    bundling: {
+  constructor(props: BundlingProps) {
+    const {
+      entry,
+      runtime,
+      architecture = Architecture.X86_64,
+      outputPathSuffix = '',
       image,
-      command: ['bash', '-c', depsCommand],
-    },
-  });
-}
-
-/**
- * Checks to see if the `entry` directory contains a type of dependency that
- * we know how to install.
- */
-export function stageDependencies(entry: string, stagedir: string): boolean {
-  const prefixes = [
-    'Pipfile',
-    'pyproject',
-    'poetry',
-    'requirements.txt',
-  ];
+    } = props;
+
+    const outputPath = path.join(AssetStaging.BUNDLING_OUTPUT_DIR, outputPathSuffix);
+
+    const bundlingCommands = this.createBundlingCommand({
+      entry,
+      inputDir: AssetStaging.BUNDLING_INPUT_DIR,
+      outputDir: outputPath,
+    });
+
+    const defaultImage = DockerImage.fromBuild(path.join(__dirname, '../lib'), {
+      buildArgs: {
```

Comment:
Can you also add a `buildArgs` property that can be passed through to here? I think it will be a common use case to just need to provide some additional buildArgs, without wanting to customize the entire docker image.

```ts
buildArgs: {
  ...props.buildArgs ?? {},
  IMAGE: runtime.bundlingImage.image,
}
```

Reply:
I considered that (and started adding it), but that'd mean adding in build args into the Docker image that can be received. Just adding them here wouldn't suffice without also updating the `Dockerfile`, so skipped it... I'm happy to add a couple standard ones for using private PyPI servers.

Reply:
Done.